src/trace/span-inferrer.ts

```diff
@@ -94,7 +94,15 @@ export class SpanInferrer {
     const options: SpanOptions = {};
     const domain = event.requestContext.domainName || "";
     const path = event.rawPath || event.requestContext.path || event.requestContext.routeKey;
-    const resourcePath = event.rawPath || event.requestContext.resourcePath || event.requestContext.routeKey;
+    var resourcePath = event.rawPath || event.requestContext.resourcePath || event.requestContext.routeKey;
```

Comment:
I'd suggest refactoring this line + 98 into a new method with your new conditional as the first path, and use early returns as control flow.

My concern would be that we shadow/overwrite `resourcePath` conditionally here in 101, and that if we add new logic here then someone needs to modify the logic on 97 they may not realize that their change would be later overwritten if it's a parameterized route.

Is that helpful?